lib/model/unittest/CResourceMonitorTest.cc

```diff
@@ -82,6 +82,25 @@ void CResourceMonitorTest::testMonitor() {
             CPPUNIT_ASSERT(false);
         }
     }
+    {
+        // Test foreground persistence constructor
+        CResourceMonitor mon(true);
+        CPPUNIT_ASSERT(mon.m_ByteLimitHigh > 0);
+        CPPUNIT_ASSERT_EQUAL((49 * mon.m_ByteLimitHigh) / 50, mon.m_ByteLimitLow);
+        CPPUNIT_ASSERT(mon.m_ByteLimitHigh > mon.m_ByteLimitLow);
+        CPPUNIT_ASSERT(mon.m_AllowAllocations);
+        LOG_DEBUG(<< "Resource limit is: " << mon.m_ByteLimitHigh);
+        if (sizeof(std::size_t) == 4) {
+            // 32-bit platform
+            CPPUNIT_ASSERT_EQUAL(std::size_t(1024ull * 1024 * 1024 / 2), mon.m_ByteLimitHigh);
```

Comment:
We haven't built 32 bit for about 5 years and I bet there are other things that would go wrong if somebody did try a 32 bit build.  So it's probably best to scrap this branch and let `sizeof(std::size_t) == 4` go into the `// Unexpected platform` branch.